plugins/editor/qml/Editor.qml

```diff
@@ -645,217 +646,231 @@ Pane{
+                        lastLength = length
+                    }
 
-                focus : true
+                    focus : true
 
-                color : "#fff"
-                font.family: "Source Code Pro, Ubuntu Mono, Courier New, Courier"
-                font.pixelSize: lk.settings.file('editor').fontSize
-                font.weight: Font.Normal
+                    color : "#fff"
+                    font.family: "Source Code Pro, Ubuntu Mono, Courier New, Courier"
+                    font.pixelSize: lk.settings.file('editor').fontSize
+                    font.weight: Font.Normal
 
-                selectByMouse: true
-                mouseSelectionMode: NewTextEdit.SelectCharacters
-                selectionColor: "#3d4856"
+                    selectByMouse: true
+                    mouseSelectionMode: NewTextEdit.SelectCharacters
+                    selectionColor: "#3d4856"
 
-                textFormat: NewTextEdit.PlainText
+                    textFormat: NewTextEdit.PlainText
 
-                wrapMode: NewTextEdit.NoWrap
-                height : Math.max( flick.height - 10, paintedHeight + 15)
-                width : Math.max( flick.width - 10, paintedWidth + 15)
+                    wrapMode: NewTextEdit.NoWrap
 
-                readOnly: editor.document === null || editor.document.isMonitored
+                    height: paintedHeight
+                    width: paintedWidth
 
-                Keys.onPressed: {
-                    if ( (event.key === Qt.Key_BracketRight && (event.modifiers === Qt.ShiftModifier) ) ||
-                         (event.key === Qt.Key_BraceRight) ){
+                    readOnly: editor.document === null || editor.document.isMonitored
 
-                        codeHandler.handleClosingBrace(cursorPosition)
-                        event.accepted = true
 
-                    } else if ( event.key === Qt.Key_PageUp ){
-                        if ( codeHandler.completionModel.isEnabled ){
-                            qmlSuggestionBox.highlightPrevPage()
-                        } else {
-                            var lines = flick.height / cursorRectangle.height
-                            var prevLineStartPos = editorArea.text.lastIndexOf('\n', cursorPosition - 1)
-                            while ( --lines > 0 ){
-                                cursorPosition   = prevLineStartPos + 1
-                                prevLineStartPos = editorArea.text.lastIndexOf('\n', cursorPosition - 2)
-                                if ( prevLineStartPos === -1 ){
-                                    cursorPosition = 0;
-                                    break;
+
+                    Keys.onPressed: {
+                        if ( (event.key === Qt.Key_BracketRight && (event.modifiers === Qt.ShiftModifier) ) ||
+                            (event.key === Qt.Key_BraceRight) ){
+
+                            codeHandler.handleClosingBrace(cursorPosition)
```

Comment:
Do we use this function? We can detect closing braces directly in the `CodeQmlHandler` during assist completion.